src/app/event/[eventId]/(eventId)/task/[taskId]/page.tsx

```diff
@@ -83,66 +167,158 @@ export default function TaskDetailPage({
+                                  <Button
+                                    variant="primary"
+                                    data-testid="button-edit-form"
+                                    onClick={() =>
+                                      handleOpenPopup(
+                                        step.id,
+                                        step.name,
+                                        step.description,
+                                        step.status,
+                                        step.step_order,
+                                        step.task
+                                      )
+                                    }
+                                  >
+                                    Edit Step
+                                  </Button>{' '}
+                                </td>
+                                <td style={{ paddingRight: '10px' }}>
+                                  {' '}
+                                  <Button
+                                    variant="primary"
+                                    onClick={() =>
+                                      handleNext(
+                                        step.id,
+                                        step.name,
+                                        step.description,
+                                        'DONE',
+                                        step.step_order,
+                                        step.task
+                                      )
+                                    }
+                                  >
+                                    {index === steps.length - 1
+                                      ? 'Finish'
+                                      : 'Continue'}
+                                  </Button>{' '}
+                                </td>
+                                <td style={{ paddingRight: '10px' }}>
+                                  {' '}
+                                  <Button
+                                    disabled={index === 0}
+                                    onClick={handleBack}
+                                    variant="ghost"
+                                    data-testid={step.id + '-back'}
+                                  >
+                                    Back
+                                  </Button>{' '}
+                                </td>
+                              </tr>
+                            </tbody>
+                          </table>
+                        </Box>
+                      </StepContent>
+                    </Step>
+                  ))}
+                  {activeStep === steps?.length && (
+                    <Paper square elevation={0} sx={{ p: 3 }}>
+                      <Typography>
+                        All steps completed - you have finished this task
+                      </Typography>
+                    </Paper>
+                  )}
+                </Stepper>
```

Comment:
I felt like this could use some refactoring, Your component line of code is too long (over 300 lines, while I believe you are trying to do it the most simple way possible but I don't think this is the best practice). Perhaps if you do refactoring, these component could also be used in the future by other features. 

Feel free to share your point of view. I would gladly accept it.

Reply:
Refactor it soon